xdsl/rewriter.py

```diff
@@ -63,6 +63,32 @@ def replace_op(
 
         block.erase_op(op, safe_erase=safe_erase)
 
+    @staticmethod
+    def inline_block_at_end(inlined_block: Block, extended_block: Block):
+        """
+        Move the block operations to the end of another block.
+        This block should not be a parent of the block to move to.
+        The block operations should not use the block arguments.
+        """
+        if inlined_block.is_ancestor(extended_block):
+            raise Exception("Cannot inline a block in a child block.")
+        for op in inlined_block.ops:
+            for operand in op.operands:
+                if (
+                    isinstance(operand, BlockArgument)
+                    and operand.block is extended_block
+                ):
+                    raise Exception(
+                        "Cannot inline block which has operations using "
+                        "the block arguments."
+                    )
+
+        ops = list(inlined_block.ops)
```

Comment:
Why do we convert this to a list? This is asymptotically bad, no?

Reply:
We need to, because we have to update all operation parents!
That's not something we can avoid sadly, MLIR has the same issue, and moving `n` operations is always `O(n)`.

Reply:
We should at least write sth like
```
for op in inline_lock.ops:
  op.detach()
```

Reply:
detach detaches from block and next/previous, so this would not work